emu_mps/utils.py

```diff
@@ -256,3 +256,26 @@ def tensor_trace(tensor: torch.Tensor, dim1: int, dim2: int) -> torch.Tensor:
     """
     assert tensor.shape[dim1] == tensor.shape[dim2], "dimensions should match"
     return tensor.diagonal(offset=0, dim1=dim1, dim2=dim2).sum(-1)
+
+
+def is_symmetric_zero_diag_matrix(
+    inter_matrix: list[list[float]] | torch.Tensor, tol: float = 1e-15
+) -> bool:
+    if len(inter_matrix) == 0:
+        return False
+
+    for column in inter_matrix:
```

Comment:
those are actually rows no?

Reply:
removed. Instead of lists now it's torch object

Reply:
I convert a list of lists to a torch object and then do checks. Waiting for Henrique's PR